pipelines/pipelines/xgboost/training/pipeline.py

```diff
@@ -184,202 +155,107 @@ def xgboost_pipeline(
             dataset_id=dataset_id,
             table_name=preprocessed_table,
             dataset_location=dataset_location,
-            file_pattern=file_pattern,
+            file_pattern="",
         )
         .after(data_cleaning)
         .set_display_name("Extract train data to storage")
-    )
+    ).outputs["dataset"]
     valid_dataset = (
         extract_bq_to_dataset(
             bq_client_project_id=project_id,
             source_project_id=project_id,
             dataset_id=dataset_id,
             table_name=valid_table,
             dataset_location=dataset_location,
-            file_pattern=file_pattern,
+            file_pattern="",
         )
         .after(split_valid_data)
         .set_display_name("Extract validation data to storage")
-    )
-    test_dataset = (
-        extract_bq_to_dataset(
-            bq_client_project_id=project_id,
-            source_project_id=project_id,
-            dataset_id=dataset_id,
-            table_name=test_table,
-            dataset_location=dataset_location,
-            file_pattern=file_pattern,
+    ).outputs["dataset"]
+
+    if test_dataset_uri:
+        test_dataset = (
```

Comment:
make sure to test both paths of this if/else

Reply:
using standard `if` statements in the pipeline definition will require recompilation if the truthiness of `test_dataset_uri` changes. i.e. you can't just change the pipeline parameter. Is this intended?